### PR TITLE
Setup Jupyter Book

### PR DESCRIPTION
**Pull request by meghanrjones**
Following the discussion at https://forum.generic-mapping-tools.org/t/proposal-creating-a-separate-gmt-tutorial-page/2831/11, this PR sets up a jupyter book to host material for the 2022 GMT for Geodesy UNAVCO short course.

Based on https://github.com/GenericMappingTools/egu22pygmt/pull/3.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: gmtforgeodesy
+channels:
+  - conda-forge
+  - nodefaults
+dependencies:
+  - gmt=6.3.0
+  - jupyterlab=3.4.0
+  - python=3.9
+  # Optional dependencies
+  - jupyter-book=0.12.3
```
Comment (weiji14) [inline environment.yml:10]: Just a note that jupyter-book released 0.13.0 yesterday (https://github.com/executablebooks/jupyter-book/releases/tag/v0.13). They've migrated from sphinx-panels to sphinx-design, so you would need to change the cards in `intro.md`. Up to you of course whether to do the migration.

```suggestion
  - jupyter-book=0.13.0
```

Comment (meghanrjones) [reply environment.yml:10]: Thanks, I will leave this comment open for now and address it as a separate issue after the initial PR due to time restrictions.

**Comment by meghanrjones** [comment]
Preview is live at https://www.generic-mapping-tools.org/gmt-for-geodesy. @GenericMappingTools/gmt-workshop-leaders, please provide feedback about using this site structure for the GMT for geodesy workshop. Currently there's little content, but all the existing bash scripts in https://github.com/GenericMappingTools/2021-unavco-course could be made into Jupyter Notebooks if we want to use this structure.

File: .github/workflows/deploy-book.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# https://jupyterbook.org/en/stable/publish/gh-pages.html#automatically-host-your-book-with-github-actions
+name: deploy-book
+
+# Only run this when the main branch changes
+on:
+  # Uncomment the 'pull_request' line below to manually re-build Jupyter Book
+  pull_request:
```
Comment (seisman) [inline .github/workflows/deploy-book.yml:7]: Need to comment this line before merging into main branch.
```suggestion
  # pull_request:
```

File: .github/workflows/deploy-book.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    - name: Checkout
+      uses: actions/checkout@v3.0.2
+
+    # Install Mambaforge with conda-forge dependencies
+    - name: Setup Mambaforge
+      uses: conda-incubator/setup-miniconda@v2.1.1
+      with:
+        activate-environment: gmtforgeodesy
+        environment-file: environment.yml
+        python-version: ${{ matrix.python-version }}
```
Comment (seisman) [inline .github/workflows/deploy-book.yml:37]: `${{ matrix.python-version }}` is not defined. Perhaps we should simply use `3.10`?
```suggestion
        python-version: 3.10
```

**Review by seisman — COMMENTED** [review]
Looks good!

**Comment by Esteban82** [comment]
I have the habit of adding a number as a prefix to the folder names (e.g. 01_introduction) so that the folders are automatically sorted. I suggest doing the same with the course material. It may be useful for students.  Here an [example](https://github.com/Esteban82/AGA-Geoinformatica-2021/tree/main/1_Ejercicios). 

For example, for the [2021 unavco](https://github.com/GenericMappingTools/2021-unavco-course) course, the names would be: 

- 00_bash
- 01_basics
- 02_line_symbols
- 03_grids, ...

**Comment by meghanrjones** [comment]
> I have the habit of adding a number as a prefix to the folder names (e.g. 01_introduction) so that the folders are automatically sorted. I suggest doing the same with the course material. It may be useful for students. Here an [example](https://github.com/Esteban82/AGA-Geoinformatica-2021/tree/main/1_Ejercicios).

Sounds good! I am not sure if or where the current first-figure example will fit into the book, so I will add the numbers at the same time as the other materials.

**Comment by seisman** [comment]
I'm closing this PR because we won't use Jupyter Notebook in this workshop.

**Pull request closed without merging**